integration_tests/src/main/python/udf_test.py

```diff
@@ -14,21 +14,22 @@
 
 import pytest
 
+pd = pytest.importorskip("pandas", reason="pandas is required for pandas UDF testing")
```

Comment:
NIT: There is also a [require_minimum_pandas_version](https://github.com/apache/spark/blob/e0538bd38cd43feaa064e30df940edf8fb2088de/python/pyspark/sql/pandas/utils.py#L19) in spark, similar to the one for pyarrow.